src/helpers.cpp

```diff
@@ -239,6 +239,16 @@ namespace ada::helpers {
       } while (true);
     }
   }
+
+  ada_really_inline void strip_trailing_spaces_from_opaque_path(ada::url& url) noexcept {
+    if (!url.has_opaque_path) return;
+    if (url.fragment.has_value()) return;
+    if (url.query.has_value()) return;
+    size_t non_whitespace_location = url.path.find_first_of(' ');
+    if (non_whitespace_location != std::string_view::npos) {
+      url.path = url.path.substr(0, non_whitespace_location);
```

Comment:
If that's correct, I recommend just doing `url.path.resize(non_whitespace_location)`. Simpler and faster.